### PR TITLE
[plug-in] Add plug-in typedoc link and comparison table of VsCode/Theia of plug-ins on website

### PR DESCRIPTION
**Pull request by benoitf**
Add useful external links for plug-ins

Change-Id: I6d39528b4d11215bac020626835ce89d442d39eb
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

**Comment by benoitf** [comment]
I think ppl will love to see api docs and https://che-incubator.github.io/vscode-theia-comparator/status.html . Let me know what is preventing to merge this PR

File: doc/Authoring_Plugins.md [hunk 1]
```diff
@@ -188,3 +188,9 @@ Then you'll only have to refresh the tab of the `Development Host` instance, the
 
 Note: you may use watch mode as well.
 
+## API of plug-ins
+[Browse typedoc of plug-ins](/api/)
+
+## VSCode implementation
```
Comment (svenefftinge) [inline doc/Authoring_Plugins.md]: It is VS Code

Comment (benoitf) [reply doc/Authoring_Plugins.md]: fixed

**Comment by svenefftinge** [comment]
Why did you delete the package-lock.json?

File: doc/Authoring_Plugins.md [hunk 2]
```diff
@@ -191,6 +191,6 @@ Note: you may use watch mode as well.
 ## API of plug-ins
 [Browse typedoc of plug-ins](/api/)
 
-## VSCode implementation
-Theia is providing VSCode API. Check the following link to get the current status of what is implemented.
+## VS Code implementation
+Theia is providing VS Code API. Check the following link to get the current status of what is implemented.
 [Compare Theia vs VsCode API](https://che-incubator.github.io/vscode-theia-comparator/status.html)
```
Comment (svenefftinge) [inline doc/Authoring_Plugins.md]: VS Code

Comment (benoitf) [reply doc/Authoring_Plugins.md]: it's already fixed, no ?

**Comment by benoitf** [comment]
> Why did you delete the package-lock.json?

@svenefftinge If I let package-lock.json, then we'll always have the same dependency to @theia/plugin then it won't be up-to-date. (we'll have only typedoc of when I updated the package-lock.json)
While I think it should always contain the latest version (and run nightly)

File: doc/Authoring_Plugins.md (hunk 2 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md]: it wasn't, now it's fixed

File: doc/Authoring_Plugins.md [hunk 3]
```diff
@@ -188,3 +188,9 @@ Then you'll only have to refresh the tab of the `Development Host` instance, the
 
 Note: you may use watch mode as well.
 
+## API of plug-ins
+[Browse typedoc of plug-ins](/api/)
+
+## VS Code implementation
+Theia is providing VS Code API. Check the following link to get the current status of what is implemented.
+[Compare Theia vs VS Code API](https://che-incubator.github.io/vscode-theia-comparator/status.html)
```
Comment (akosyakov) [inline doc/Authoring_Plugins.md]: Is it based on d.ts files comparison? I know that we have something in theia.d.ts but they are not used or implemented, i.e. `FileSystemProvider` except read and write, nothing works. It is marked as green.

Comment (akosyakov) [reply doc/Authoring_Plugins.md]: but super cool idea

**Comment by svenefftinge** [comment]
I think the docs should be published by the theia repo/build. Otherwise it won't be automatically updated if we release. Also removing the lock is not good as it potentially breaks our build without us doing anything.

File: doc/Authoring_Plugins.md (hunk 3 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md]: @akosyakov yes it's only d.ts comparison, it doesn't check implementation

if it's green it doesn't mean it's compliant, but if it's red, surely it doesn't work

**Comment by benoitf** [comment]
@svenefftinge how to do that during theia repo/build ? as website is managed in this repository ?

about the lock , I may remove carets for other dependencies to stabilize it a very little more.

**Comment by svenefftinge** [comment]
> @svenefftinge how to do that during theia repo/build ? as website is managed in this repository ?

we need to make sure the api docs get generated and published on every release, so they match the respective versions on npm (incl. latest and next). I wonder if there is not already a tool that does that. Sounds too obvious.

File: doc/Authoring_Plugins.md (hunk 2 above)
Comment (svenefftinge) [reply doc/Authoring_Plugins.md]: In the link description it is not.

**Comment by svenefftinge** [comment]
The website can then just link to 'latest'.

**Comment by benoitf** [comment]
@svenefftinge is it ok to publish typedoc to github pages of theia-ide project ? (gh-pages branch)

**Comment by svenefftinge** [comment]
Yes, I think that should be fine.

**Comment by benoitf** [comment]
ok I will rewrite this PR and open another one to https://github.com/theia-ide/theia repository

**Comment by svenefftinge** [comment]
Thank you!

**Comment by benoitf** [comment]
@svenefftinge : https://github.com/theia-ide/theia/pull/4636

**Comment by benoitf** [comment]
@svenefftinge updated

**Comment by benoitf** [comment]
is it ok now ? it's two new links added

**Pull request merged**